src/Collapse.js

```diff
@@ -55,34 +55,46 @@ class Collapse extends Component {
       height: null
     };
 
+    this.nodeRef = React.createRef();
+
     ['onEntering', 'onEntered', 'onExit', 'onExiting', 'onExited'].forEach((name) => {
       this[name] = this[name].bind(this);
     });
   }
 
-  onEntering(node, isAppearing) {
+  getNode() {
+    const nodeRef = this.props.innerRef || this.nodeRef;
+    return nodeRef.current;
+  }
```

Comment:
This would need to consider callback refs as well.  Also, looking at the proptypes, it looks like it also allows legacy refs.

Reply:
I will look into supporting callback refs.

As for legacy refs, should I check if the type of `props.innerRef` is a string, and add a workaround for it?

Reply:
If we do the major version bump, I think we should drop legacy refs.

Reply:
The new RTG `nodeRef` API only supports object refs (I've tried passing it a function ref and it triggered a proptypes warning).

I've updated the prop types on `Collapse` to only allow object refs. Is this a good solution?

Reply:
Ugh... didn't realize it only took ref objects.  We should still allow both callback and object refs to be passed in through `innerRef` though... Unfortunately I only know of how to merge refs via hooks.  Unsure what approach to take here.